### DISCUSSION TITLE
Question about D3JSVisual (forecast tab)

### DISCUSSION THREAD

**Discussion by Drammed** [nbrown02/FlowViz-Jira#40] [Q&A] [answered]
Dear team, 

We have discovered FlowViz and we are superhappy with it, so thank you for the work you have done to create and share this! 
Hopefully you are able to guide us to a solution for our implementation regarding the D3JSVisual on the forecast tab. 

**Situation:**
- Our company policy is blocking all non-Microsoft-certified visuals
- FlowViz is using D3JSVisual for the Forecast tab. This is not a certified visual. 
- Therefore we get the following error after publication. This reduces the value of FlowViz for our company: 
![image](https://github.com/nbrown02/FlowViz-Jira/assets/173803564/14a34e1e-1f4e-4bb0-b3c8-e322e9928d59)

**Questions**
1. Is there any way we can work around this error and use the Forecast tab without enabling the uncertified D3JSVisual to the entire company? 
2. If not, is there any chance this visual can get certified? 
3. If not, is there any way we can prove to our security department that if we enable the D3JSVisual that it...
a ...runs locally and does not extract information (that could potentially leak at some point in time)? 
b ...we can in any way limit the number of report builders that has access to this visual?

**Comment by nbrown02** [comment]
Hey there - thanks for reaching out!

So to answer your questions:

1. The only way would be using it locally in Power BI Desktop
2. It's not my visual, you'd have to ask https://github.com/liprec/powerbi-d3jsvisual
3. Same as question 2 

Sorry I can't be more help, I don't have the skillset to rebuild it myself in a non-D3 visual, you are welcome of course to fork it and create your own for the community :)

**Reply by Drammed** [reply]
Alright, thank you for the swift response! If our investigation returns interesting results for the community, I'll share them here.

**Reply by nbrown02** [reply, marked as answer]
@Drammed in the latest version I have changed the forecast visual to be a native visual bar chart rather than a d3 visual. I'm pretty certain this will fix any issues you were having :)
